src/callback.rs

```diff
@@ -0,0 +1,273 @@
+// Copyright (C) 2020-2023 Cisco Systems, Inc. and/or its affiliates. All rights reserved.
+//
+// This program is free software; you can redistribute it and/or modify
+// it under the terms of the GNU General Public License version 2 as
+// published by the Free Software Foundation.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program; if not, write to the Free Software
+// Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston,
+// MA 02110-1301, USA.
+
+use crate::{engine::ScanEvent, layer_attr::LayerAttributes, ContentHandle, EngineError};
+use clamav_sys::cl_error_t;
+use std::{
+    ffi::CStr,
+    io::Cursor,
+    os::raw::{c_char, c_int, c_uchar, c_void},
+    pin::Pin,
+};
+use tokio::io::AsyncRead;
```

Comment:
It's not a convention I've seen in many crates.  I used to do the same, but stopped since my IDE (VSCode) automatically adds dependencies as I use them, and tends to get confused if they were placed in separate groups (e.g., it will make yet another group).

`cargo fmt` *will* float your `crate::` imports to the top of the list, which is close. But it blends the `std::` imports alphabetically along with the 3rd-party dependencies.

The [style guide](https://doc.rust-lang.org/nightly/style-guide/items.html) doesn't appear to address this issue... yet.